lib/moonshot/stack.rb

```diff
@@ -176,6 +188,25 @@ def add_parameter_overrides(hash)
 
     private
 
+    def asgs
+      resources_of_type('AWS::AutoScaling::AutoScalingGroup')
+    end
+
+    def instance_ip(instance_id)
+      Aws::EC2::Client.new.describe_instances(instance_ids: [instance_id])
+                      .reservations.first.instances.first.public_ip_address
+    end
+
+    def instances
+      result = []
+      Aws::AutoScaling::Client.new.describe_auto_scaling_groups(
+        auto_scaling_group_names: asgs.map(&:physical_resource_id)
+      ).auto_scaling_groups.each do |asg|
+        result += asg.instances.map(&:instance_id)
```

Comment:
I would just stick with the functional style in this method:
```ruby
def instances
  client = Aws::AutoScaling::Client.new
  client.describe_auto_scaling_groups(auto_scaling_group_names: asgs.map(&:physical_resource_id))
    .map { |asg| asg.instances.map(&:instance_id) }
    .flatten
end
```

That being said if we go with the "you must specify the ASG if you have more than one" suggestion I made in the PR, this method would become much simpler anyway.